C2GUILauncher/src/HttpHelpers.cs

```diff
@@ -19,6 +20,11 @@ static class HttpHelpers {
         public static DownloadTask DownloadFileAsync(string url, string outputPath) {
             if (!Directory.Exists(Path.GetDirectoryName(outputPath)))
                 Directory.CreateDirectory(Path.GetDirectoryName(outputPath)!);
+            else if (IsFileLocked(outputPath))
+                return new DownloadTask(
+                Task.CompletedTask,
+                new DownloadTarget("", null)
```

Comment:
This should just return the `url` and `outputPath` that were passed in to the function rather than null, if the idea of a lock is to just prevent a download and preserve all other behavior.

As it is now, this would need updating: https://github.com/Chiv2-Community/C2GUILauncher/blob/d3e9e04480244be5b496b491185bdc76b07c91a3/C2GUILauncher/src/ModManager.cs#L223

The reference to `OutputPath!` would throw an NPE.